### PR TITLE
azure: Allow user to set custom data

### LINKED ISSUE #3968: Support Injection of Custom Data into the ARM Template
Azure allows users to specify [custom data][1], which can be used to influence the cloud-init process.  The Azure builder should allow users to specify custom data.

[1]: https://azure.microsoft.com/en-us/documentation/articles/virtual-machines-linux-using-cloud-init/

### PR DESCRIPTION
**Pull request by boumenot**
Closes #3968

### CODE REVIEW

**Comment by boumenot** [comment]
PTAL: @paulmey

File: CHANGELOG.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ IMPROVEMENTS:
   * builder/azure: Support for custom images [GH-3575]
   * builder/azure: Removed superfluous polling code for deployments [GH-3638]
   * builder/azure: Made `tenant_id` optional [GH-3643]
+  * builder/azure: Add `os_disk_size_gb` [GH-3965]
```
Comment (paulmey) [inline CHANGELOG.md]: you probably want to just commit this to the changelog? seems unrelated

File: builder/azure/arm/config_test.go [hunk 2]
```diff
@@ -673,6 +673,29 @@ func TestConfigShouldRejectExcessiveTagValueLength(t *testing.T) {
...
+		"subscription_id":        "ignore",
+		"communicator":           "none",
+		// Does not matter for this test case, just pick one.
+		"os_type":          constants.Target_Linux,
+		"custom_data_file": "/this/file/does/not/exist",
+	}
+
+	_, _, err := newConfig(config, getPackerConfiguration())
+	if err == nil {
+		t.Fatal("expected config to reject tag value based on length (> 256)")
```
Comment (paulmey) [inline builder/azure/arm/config_test.go]: this doesn't look like the right error message for this test

File: CHANGELOG.md (hunk 1 above)
Comment (boumenot) [inline CHANGELOG.md]: Removed.

File: builder/azure/arm/config_test.go (hunk 2 above)
Comment (boumenot) [inline builder/azure/arm/config_test.go]: Good catch!

**Comment by mwhooker** [comment]
👍
